src/logic/safe/hooks/useLoadSafe.tsx

```diff
@@ -24,12 +25,16 @@ export const useLoadSafe = (safeAddress?: string): void => {
           .then(() => {
             dispatch(fetchSafeCreationTx(safeAddress))
             dispatch(fetchTransactions(safeAddress))
-            return dispatch(addViewedSafe(safeAddress))
+            dispatch(addViewedSafe(safeAddress))
+            setSafeLoaded(true)
```

Comment:
@Agupane, help me understand this.

How I see it is:

```ts
dispatch(fetchLatestsMasterContractVersion())
  // here we're waiting for `fetchLatestsMasterContractVersion` to resolve
  .then(() => {
     // after the previous promise resolution we dispatch two different actions
     dispatch(fetchSafe(safeAddress)) // !!! - I understand this as the most important
     return dispatch(fetchSafeTokens(safeAddress))
  })
  // we wait for `fetchSafeTokens` action to resolve
  .then(() => {
    /*
      ...
    */
    setSafeLoaded(true) // !!! - basically, we're waiting for `fetchSafeTokens` to resolve to switch the flag
    return
  })
```

How I see it (and how I understand it), we may face a race condition if `fetchSafe` takes too long to resolve.

Isn't it?

Reply:
As far as I investigated yes but it's not always simple to get that status

Reply:
I updated the code to be sure that `fetchSafe` is executed in order to update the flag, seems that the problem is with multiple redux actions executing at the same time, at least at know, is the best solution I found